src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -2,89 +2,77 @@
 
 import java.util.Collections;
 import java.util.List;
+import java.util.NoSuchElementException;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 public class JavaStreamApi {
 
-    /**
-     * <p>1. Дано: List of Integer numbers.
-     * Вернуть сумму нечетных числел или 0, если таких несуществует</p>
-     **/
     public Integer oddSum(List<Integer> numbers) {
-        return null;
+        return numbers.stream().filter(number -> number % 2 == 1).mapToInt(Integer::intValue).sum();
     }
 
-    /**
-     * <p>2. Дана коллекция строк List of String elements
-     * (пример: Arrays.asList(«a1», «a2», «a3», «a1»)).
-     * Вернуть количество вхождений объекта `element`</p>
-     **/
     public Long elementCount(List<String> elements, String element) {
-        return null;
+        return elements.stream().filter(s -> s.equals(element)).count();
     }
 
-    /**
-     * <p>3. Дана коллекция строк List of String elements
-     * (пример: Arrays.asList(«a1», «a2», «a3», «a1»)).
-     * Вернуть Optional первого элемента коллекции</p>
-     **/
     public Optional<String> firstElement(List<String> elements) {
-        return null;
+        return elements.isEmpty() ? Optional.empty() : Optional.ofNullable(elements.get(0));
```

Comment:
https://mate-academy.github.io/jv-program-fulltime/01_core/checklist/hw_19.html

Reply:
use stream to solve this task. You should find the right  solution but not just make the tests passed

Reply:
done